packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -245,11 +295,48 @@ export default {
     },
     filteredAppMap() {
       const { appMap } = this.$store.state;
-      const events = appMap.rootEvents().reduce((callTree, rootEvent) => {
+      let rootEvents = appMap.rootEvents();
+
+      if (this.filters.limitRootEvents.on) {
+        rootEvents = rootEvents.filter((e) => e.httpServerRequest);
+      }
+
+      let events = rootEvents.reduce((callTree, rootEvent) => {
         rootEvent.traverse((e) => callTree.push(e));
         return callTree;
       }, []);
 
+      events = events.filter(
+        (e) => this.filters.unlabeledEvents.on || e.labels.size > 0
+      );
+
+      if (this.filters.hideMediaRequests.on) {
+        const mediaTypes = [
+          'application/javascript',
+          'application/ecmascript',
+          'audio/.+',
+          'font/.+',
+          'image/.+',
+          'text/javascript',
+          'text/ecmascript',
+          'text/css',
+          'video/.+',
+        ];
+        const excludedEvents = [];
+        events.forEach((e) => {
+          if (
+            e.http_server_response &&
+            e.http_server_response.mime_type &&
+            mediaTypes.some((type) =>
+              new RegExp(type, 'ig').test(e.http_server_response.mime_type)
```

Comment:
Compiling the regex for each iteration of this loop is quite slow. Can we compile these once?
https://jsbench.me/tnkph4i56y/1

Reply:
Good point. Done https://github.com/applandinc/appmap-js/pull/217/commits/4c26a606138eb215a3cd4b7d3ff78bf88c329d54